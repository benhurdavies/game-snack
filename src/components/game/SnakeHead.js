import React from 'react';
import { Shape } from 'react-konva';

const SnakeHead = ({ x, y, width, height, blockSize, theme }) => (
  <Shape
    x={x * blockSize}
    y={y * blockSize}
    fill={theme.snakeColor}
    sceneFunc={(ctx, shape) => {
      ctx.beginPath();
      ctx.moveTo(0, 0);
      ctx.lineTo(10, 0);
      ctx.bezierCurveTo(20, 0, 20, 20, 10, 0);
      //context.lineTo(20, 20);
      ctx.lineTo(0, 10);
      ctx.lineTo(0, 0);
      ctx.stroke();
    }}
  />
);

export default SnakeHead;
