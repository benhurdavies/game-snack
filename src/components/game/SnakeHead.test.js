import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import SnakeHead from './SnakeHead';

describe('Component: SnakeHead', () => {
  let props = {
    x: 10,
    y: 10,
    blockSize: 20,
    theme: { snakeColor: 'black', snakeColorInvert: 'white' },
    towards: 'left',
  };

  it('basic shape check', () => {
    const wrapper = shallow(<SnakeHead {...props} />);
    expect(wrapper.find('Shape').length).toEqual(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<SnakeHead {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
