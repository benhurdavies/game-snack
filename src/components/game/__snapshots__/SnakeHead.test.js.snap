// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SnakeHead renders correctly 1`] = `
<Shape
  sceneFunc={[Function]}
  x={200}
  y={200}
/>
`;
